kuma/javascript/src/payments/pages/management.jsx

```diff
@@ -0,0 +1,200 @@
+            <Interpolated
+                id={gettext(
+                    'You have no active subscriptions. Why not <signupLink />?'
+                )}
+                signupLink={
+                    <a href={`/${locale}/payments/`}>{gettext('set one up')}</a>
+                }
+            />
+        </div>
+    );
+
+    const renderSubscriptions = () => {
+        if (!subscription) {
+            return false;
+        }
+        const date = new Date(subscription.next_payment_at);
+        const nextPaymentDate = formatDate(
+            locale,
+            subscription.next_payment_at
+        );
+        const lastActiveDate = formatDate(
+            locale,
+            date.setDate(date.getDate() - 1)
+        );
+        return (
+            <>
+                <p>
+                    {interpolate(
+                        'Next payment occurs on %(nextPaymentDate)s.',
+                        {
+                            nextPaymentDate,
+                        }
+                    )}
+                </p>
+                <div className="active-subscriptions">
+                    <ul>
+                        <li>
+                            <span className="label amount">
+                                {gettext('Amount')}
+                            </span>
+                            <span className="value">
+                                {/* amount is in cents, so divide by 100 to get dollars */}
+                                {formatMoney(locale, subscription.amount / 100)}
+                            </span>
+                        </li>
+                        <li>
+                            <span className="label credit-card">
+                                {gettext('Card number')}
+                            </span>
+                            <span className="value">{`**** **** **** ${subscription.last4}`}</span>
+                        </li>
+                        <li>
+                            <span className="label credit-card">
+                                {gettext('Expiry')}
+                            </span>
+                            <span className="value">
+                                {subscription.expires_at}
+                            </span>
+                        </li>
+                    </ul>
+                </div>
+                <button
+                    className="cta cancel"
+                    onClick={handleClick}
+                    type="button"
+                >
+                    {gettext('Cancel subscription')}
+                </button>
+                {showForm && (
+                    <CancelSubscriptionForm
+                        onSuccess={handleDeleteSuccess}
+                        setShowForm={setShowForm}
+                        date={lastActiveDate}
+                    />
+                )}
+            </>
+        );
+    };
+
+    const renderContent = () => {
+        switch (true) {
```

Comment:
Another curiosity question :) This is an interesting use of the `switch` statement, is there a specific benefit to this over using `if/else`

Reply:
I'm indifferent. Personally *I* prefer plain `if/else if` blocks but I don't think you should change it if you prefer this style here.

Reply:
I usually use an if/else unless it's more than 3 blocks, then I use a `switch` statement for faster performance (this is what I have read, I've never actually done a speed test comparing the two).

Reply:
Thank you for the feedback Mindy.